src/System.Management.Automation/engine/CommandProcessorBase.cs

```diff
@@ -87,6 +91,11 @@ internal bool AddedToPipelineAlready
         /// <value></value>
         internal CommandInfo CommandInfo { get; set; }
 
+        /// <summary>
+        /// Gets whether the command has a 'Clean' block defined.
+        /// </summary>
+        internal bool HasCleanBlock { get; }
```

Comment:
I think the `Cleanup` block should be renamed to `Clean` block. `Begin/Process/End` are all verbs, and mapping to `BeginProcessing/ProcessRecord/EndProcessing`. So I think this new block should be mapped to `CleanResource`, and have `Clean` as its name. Of course, this will be a decision from the committee. I will make sure the name is consistent once it's decided what we will use.